Code/src/main/java/nl/utwente/group10/ui/components/blocks/FunctionBlock.java

```diff
@@ -76,6 +83,8 @@ public FunctionBlock(String name, Type type, CustomUIPane pane) {
 
         this.name = new SimpleStringProperty(name);
         this.type = new SimpleStringProperty(type.toHaskellType());
+        this.bowtieIndex = new SimpleIntegerProperty();
+        bowtieIndex.addListener(this);
```

Comment:
Makkelijker is misschien om hier gewoon dit te doen:

    bowtieIndex.addListener(e -> invalidateBowtieIndex());

Dan hoef je niet meer `ChangeListener` te implementeren en heb je `changed(...)` niet meer nodig.